Security Updates

BalajiJBcs opened the pull request:
@Zachary-Kuhn , I'm sorry to bring this to your concern. I see the following file is not supposed to be here. 
I don't know whether it as some use in Code.gov. But as far my knowledge it is a small human error where you saved your personnel file which everyone does. So I removed the file. If I understand wrongly and removed that pdf. Pardon me for it Mr.Zachry-Kuhn.

IanLee1521 committed:
Updated sample code.json file to match updates to metadata schema

Based on discussion in #187 / #196

IanLee1521 committed:
Updated field types per http://json-schema.org

integer -> number
list -> array

Commit:
updated metadata schema specification with changes

The following changes aim to clarify the requirements for compliance with
version 1.0.1 of the code.gov metadata schema:

Schema version: The first official iteration of the the code.gov metadata
schema will be version 1.0.1. Adding a version number to the schema makes
it easier to track and manage changes between versions of the metadata
schema.

‘Organization’ field: The ‘organization’ field, which holds the name of the
sub-agency responsible for a particular repo or project, will now (1) be an
optional field and (2) move into the ‘projects’ object. This enables
agencies to identify the appropriate organization within that agency that
owns the repository, if such an organization exists.

‘Projects’ field: The code.gov policy guide previously referred to a
‘project’ field in the metadata schema, while the sample code.json referred
to the plural from ‘projects’. The policy guide will be updated to refer to
the plural form ‘projects’.

‘govwideReuseproject’ field: The code.gov policy guide previously referred
to ‘governmentWideReuseProject’ field in the metadata schema, while the
sample code.json referred to ‘govwideReuseproject’. The sample code.json
will be updated to refer to ‘‘governmentWideReuseProject’ in order to
conform with the code.gov policy guide.

‘tags’ field: The code.gov policy guide and the sample json refer to a
‘tags’ field, defined as an array of strings. However, previous iterations
of the metadata schema and the sample code referred to a ‘projectTags’
field, which was structured as an array of ‘tag’ objects. To clarify, for
version 1.0.1, we intend to define the 'tags’ field as an array of strings
for simplicity and consistency with the ‘languages’ field.

‘exemption_text’ field: This field allows agencies to provide a brief
narrative explanation for the exception requested in the 'exemption' field.

Commit:
sample code.json for schema v1.0.1

Commit:
added reference to 'governmentWideReuseProject'

Commit:
made 'repository a required field'

Commit:
updating the description of 'partners' field

Commit:
converted 'projects' to array; clarified partners

listed out name and email in partners description

Commit:
remove _draft_content folder

* remove confusing/extraneous policy information

mattbailey0 committed:
formating and language tweaks. add changelog

mattbailey0 committed:
Merge branch 'ok-schema-fix'

Commit:
updated link to sample code.json file

michael-balint committed:
Added `name` field back.

michael-balint committed:
Moved `organization` into `projects`.

michael-balint committed:
Refactored `exemption_text` --> `exemptionText`.

michael-balint committed:
Formatting changes.

Commit:
updated exemption_text ->exemptionText

for consistency

michael-balint committed:
Refactored `exemption_text` --> `exemptionText` in the sample code.json.

mattbailey0 committed:
wording tweaks for schema content

mattbailey0 committed:
resolve merge conflicts

Commit:
remove compliance metadata schema folder

lukad03 committed:
Fix Selenium Deployment Issues

lukad03 committed:
Remove ComplianceMetaDataSchemaComponent from Imported Components

lukad03 committed:
Remove trailing comma in tsconfig

lukad03 committed:
Remove ComplianceMetaDataSchema Routes

IanLee1521 committed:
Updated type of `projects` in schema from object -> array

Commit:
added section for compliance dashboard

Commit:
new dashboard style

Commit:
added federalist branch

Commit:
update package.json

Commit:
Merge branch 'ok-compliancedash'

mattbailey0 committed:
this has been bugging me for a while

EtherTyper committed:
Update repos.json

OpenMCT is licensed under Apache v2.0.

EtherTyper committed:
Fix Apache license in repo_license.json

Commit:
added color circles to dashboard

Commit:
updated compliance dashboard circles

Commit:
add federalist staging for deploy

Commit:
fix broken css in compliance dashboard

yozlet committed:
Undo repository.repoURL rename

lukad03 committed:
Create SearchService, ReposSearchComponent, and SearchResultsComponent (#202)

* Create SearchService, ReposSearchComponent, and SearchResultsComponent

Creates various Components and a SearchService for adding basic search functionality to Code.gov.

* Create SearchService with predefined search arguments for returning data from the Code.gov API
* Create ReposSearchComponent for handling query submissions to the SearchService
* Add ReposSearchComponent to Homepage Banner
* Create SearchResultsComponent for returning SearchResults

* Connect Repo Page to API

Creates a RepoService for handling interactions between the RepoComponent and the Code.gov API.

* Create RepoService to fetch a given Repo from the API
* Modify RepoComponent to use RepoService to fetch RepoData
* Redirect /repos/:id to 404 page if Repo does not exist
* Create explicit 404 route
* Add tests for the RepoSearchComponent
* Fix CircleCi tests by forcing 2.1.1 for TypeScript in package.json

* Add Infinite Scroll to Search Results

Loads additional results as Users scroll through the SearchResults page.

* Add angular2-infinite-scroll package for handling infinite scroll
* Add infinite-scroll Directive to SearchResultsComponent Template
* Modify SearchService to store Results
* Add variables to SearchResultsComponent to update based on SearchService changes

* Use URL Params for Search

Converts search functionality to be URL param-based, as opposed to depending on the SearchService.

* Simplify SearchService
* Add Search Route
* Use queryParams to store Search queries
* Add Input to RepoSearchComponent to pass URL query param to it

* Create Load More Results Button

Creates a Load More Results Button by loading results in the background and displaying the results when a User clicks the button.

* Add Load More Results Button to SearchResultsComponent
* Modify inifinite scroll to work with Load More button

* Add Mobile Styles and Results Content

Adds Mobile Styles to SearchResultsComponent and displays the total number of results at the top of the page.

* Add Mobile Styles to SearchResultsComponent
* Display total number of results from query in the SearchResultsComponent

* Create RepoListItemComponent

Refactors lists of Repos in AgencyComponent and SearchResultsComponent to use a RepoListItemComponent.

* Create RepoListItemComponent
* Use RepoListItemComponent in AgencyComponent and SearchResultsComponent to DRY up templates

* Create RepoListComponent

Creates a RepoListComponent to refactor Agency and SearchResults Components.

* Create RepoListComponent
* Implement RepoListComponent in the SearchResultsComponent

* Use RepoListComponent in AgencyComponent

Adds the RepoListComponent to the AgencyComponent template to handle listing of Agency repos.

* Use RepoListComponent in AgencyComponent template to DRY up repo listing implementation
* Remove unnecessary functions in the AgencyComponent
* Remove ReposService

* Add ReposSearch to Explore Code Subnav

Replaces the "By Agency" link with a Search Bar to encourage Users to Search for code.

* Add ReposSearchComponent to ExploreCodeComponent subnav to enable Search in /agencies/:agency and /repos/:repo.

* Reorganize Components and Create Explore Code Page

Reorganizes the Agency and Repo Components to no longer be nested under Explore Code (per the change in routing). Also creates a basic Explore Code page.

* Move Agency-related Components to root of Components folder
* Move Repo-related Components to root of Components folder
* Add styles and content to ReposComponent
* Refactor routes into individual files for each section of the site

* Create Api and Term Services

Creates an abstracted ApiService for fetching data from the Code.gov API and a TermService for hitting the `/terms` endpoint with a query.

* Create abstracted ApiService for generalized API calls
* Create TermService that extends the ApiService for calls to the `/terms` endpoint

* Create AutocompleteComponent

Creates an AutocompleteComponent for fetching results while inputing a query into the RepoSearchComponent.

* Create AutoCompleteComponent

* Create AutocompleteResultComponent

Creates an AutocompleteResultComponent for displaying the Results from the `/terms` endpoint.

* Create AutocompleteResultComponent
* Add styles to AutocompleteComponent

* Return Agency Data from API via Autocomplete

Modifies the AgencyService, TermService, and AutoComplete-related Components to enable Agency data to be returned via Autocomplete.

* Modify AgencyService to use API
* Modify AutoCompleteResult to lookup Agency data from API
* Store Agency data to prevent unnecessary external calls to the API
* Modify TermService to store previously found Terms (prevents duplicate Results in Autocomplete)
* Modify RepoSearchComponent to show and hide the AutocompleteComponent based on Blur, Focus, and Query presence
* Add pending specification to tests where appropriate
* Add stlyes to Autocomplete and Autocomplete Result

* Modify Staging Config for Search Deployment

* Refactor AgencyComponent to Use AgencyService

Refactors the Agency View to use the AgencyService.

* Use AgencyService in AgencyComponent for populating Agency data

* WIP

Commit:
updated staging/production deployment rules

*github-deploy loads ‘dev’ webpack by default
*github-deploy:stag loads ‘federalist-stag’ web pack and pushes to
federalist-pages

*github-deploy:prod loads ‘prod’ web pack and pushes to federalist-pages

Commit:
updating deployment script to use federalist

Commit:
adding note about clearing cache

Commit:
updated federalist deploy script

*federalist-deploy:dev now uses webpack.prod.js

Commit:
Revert "Create SearchService, ReposSearchComponent, and SearchResultsComponent" (#232)

* Revert "adding note about clearing cache"

This reverts commit afc909a93bd0f616918790e32214f1550ec011a2.

* Revert "Create SearchService, ReposSearchComponent, and SearchResultsComponent (#202)"

This reverts commit e4fd86037df729cf4d98a1905357a31f7216c43d.

Commit:
added federalist dashboard preview

Commit:
adding node-sass to resolve compiling errors?

Commit:
comment out compliance dashboard

Commit:
adding federalist deployment fixes

Commit:
updated baseURL for production

Commit:
updating federalist production script

Commit:
Ok ossconmaster (#244)

* retire whats-new

* rough draft for agency compliance dashboard

* style and content tweaks to dashboard

* don't include agencies that don't need to comply in the dash

* a lil' decrufting

* updating dashboard status

* update federalist deployment script for production

* updating compliance dashboard

* updating compliance dashboard (NASA->green)

* updating dashboard

* updated list of agencies with OS or gov-wide reuse repos

* added 2 DOD repos

* updated repo template to use 'Contact Agency' and agency index template to use 'Government-Wide Reuse'

* updated repo template for govwidereuse highlights; added new DOD repos;

* add mailing list subscribe link/button

* changed order of DOD repos

* Added DOT, NRC logos and repos; added mailchimp bitly url

* added pshtt to DHS repos

* added new banner branding; changed width of branding box

* remove external link notification for mailchimp

* updating compliance dashboard

* changed mailing list button to text

* updated NRC - Adaptive Case Management

* updating compliance dashboard

* updating compliance dashboard

* updating repository component test

* updating compliance dashboard

* update test suite to reflect new repo format from API (#243)

* update test suite to reflect new repo format from API

* adding updated repo.template.html

* updated dashboard

* fix DOT dashboard

* update DOT dashboard

* fixing repo template

Commit:
fix compiling errors

Commit:
remove conflict markers

Commit:
add agency logos

Commit:
updated analytics reporter for GSA

Commit:
update compliance dashboard

Commit:
adding NSA repos (#245)

Commit:
adding FDA to agency listing (#246)

Commit:
updated DOL complance (#247)

PhilipBale committed:
Add licensing and software procurement pages to policy (#249)

* Add licensing and software procurement pages to policy.  Also move “measuring code” to the open-source menu listing.

* Style updates per Alvand’s comments

Commit:
fixing DOL score

Commit:
updating DOL score (#251)

Commit:
updating dashboard with weekly data (#252)

Commit:
Pb include source code policy (#253)

* Creation of policy files (introduction and objectives) to include policy on code.gov website

* Scope & applicability, solutions analysis, and code-reuse sections.

* Open source, exceptions, implementation, and appendix pages and page anchors internally

* Share CSS for documentation pages

* Newlines

* Lint

Commit:
Revert "Pb include source code policy (#253)"

This reverts commit 05217b62785e68d5c4499b869d99b8c786b3c6c5.

Commit:
weekly dashboard update (#255)

Commit:
updating DOL & DOT scans (#258)

Commit:
dashboard update with new DOL score (#259)

Commit:
dashboard update (#260)

Commit:
adding 18faccessibility (#262)

Commit:
updating weekly compliance dashboard (#265)

zachary-kuhn committed:
Fix linting errors and upgrade dependencies. (#266)

PhilipBale committed:
Pb include source code policy #33 (#256)

* Creation of policy files (introduction and objectives) to include policy on code.gov website

* Scope & applicability, solutions analysis, and code-reuse sections.

* Open source, exceptions, implementation, and appendix pages and page anchors internally

* Share CSS for documentation pages

* Newlines

* Lint

* nvmrc file

* Update packages.

* Update to the new Neat API and add pathMatch to redirect routes.

zachary-kuhn committed:
Merge remote-tracking branch 'origin/master' into EtherTyper-patch-1

zachary-kuhn committed:
Merge pull request #267 from presidential-innovation-fellows/EtherTyper-patch-1

Update licensing for OpenMCT

zachary-kuhn committed:
Upgrade Circle CI config to version 2.0 (#268)

zachary-kuhn committed:
Merge pull request #220 from IanLee1521/patch-3

Updated type of `projects` in schema from object -> array

zachary-kuhn committed:
Implement updated styleguide across the code.gov site.

Commit:
weekly update (#270)

zachary-kuhn committed:
Implement style guide [WIP] (#271)

* Initial work on standardizing the styles used within the app.

* Create a sidebar component.

* Add subnav and sidebar components.

Commit:
Revert "Implement style guide [WIP] (#271)" (#272)

This reverts commit 3811832838382d8625b8c9a04bda4137b2b7409f.

zachary-kuhn committed:
Merge pull request #274 from presidential-innovation-fellows/zk-implement-style-guide

Implement Style Guide

zachary-kuhn committed:
Template changes to follow SEO best practices.

zachary-kuhn committed:
Merge pull request #275 from presidential-innovation-fellows/zk-structural-changes-for-seo

Template changes to follow SEO best practices.

Commit:
updating dashboard (#277)

* updating dashboard

* fixing ok weekly update

zachary-kuhn committed:
Improve social graph meta tags (#276)

* Initial work on improving social graph tags for repos.

* Add social tags for Agencies.

zachary-kuhn committed:
Change settings for the production build and also minify CSS.

Commit:
Ok new metadataschema (#279)

* started draft schema and sample

* revised version 200

*changed projects to releases
*added field descriptions
*added licenses object
*added permissions object
*added laborHours
*revised measurementType

* removed extra " in schema; formatted diff

* revised exemptionText definition

zachary-kuhn committed:
Add the sitemap.xml and robots.txt.

zachary-kuhn committed:
Merge pull request #282 from presidential-innovation-fellows/zk-add-site-meta-information

Add the sitemap.xml and robots.txt.

zachary-kuhn committed:
Merge pull request #278 from presidential-innovation-fellows/zk-minify-css-and-use-angular-prod-build

Change settings for the production build and also minify CSS.

zachary-kuhn committed:
Fix accessibility issues (#284)

* Fix the accessibility issues found in the WCAG and Section 508 audits.

* Tweak display after design review.

Commit:
fix draft schema

Commit:
Ok new metadataschema (#286)

* started draft schema and sample

* revised version 200

*changed projects to releases
*added field descriptions
*added licenses object
*added permissions object
*added laborHours
*revised measurementType

* removed extra " in schema; formatted diff

* revised exemptionText definition

* updated schema based on github comments

*moved measurementType outside of ‘releases’
*added reusedSourceCode object

* changed reusedSourceCode to relatedCodebase

* added changelog.md

* removed 'hourstoAdopt'

* removed 'hourstoadopt' from diff

* Restrict JSON schema to not include unspecified properties.

* Fix sample Code.json to match schema.

Commit:
Ok new metadataschema (#287)

* started draft schema and sample

* revised version 200

*changed projects to releases
*added field descriptions
*added licenses object
*added permissions object
*added laborHours
*revised measurementType

* removed extra " in schema; formatted diff

* revised exemptionText definition

* updated schema based on github comments

*moved measurementType outside of ‘releases’
*added reusedSourceCode object

* changed reusedSourceCode to relatedCodebase

* added changelog.md

* removed 'hourstoAdopt'

* removed 'hourstoadopt' from diff

* Restrict JSON schema to not include unspecified properties.

* Fix sample Code.json to match schema.

* update description of laborHours

* add new metadata schema and changelog

Commit:
Ok new metadataschema (#288)

* started draft schema and sample

* revised version 200

*changed projects to releases
*added field descriptions
*added licenses object
*added permissions object
*added laborHours
*revised measurementType

* removed extra " in schema; formatted diff

* revised exemptionText definition

* updated schema based on github comments

*moved measurementType outside of ‘releases’
*added reusedSourceCode object

* changed reusedSourceCode to relatedCodebase

* added changelog.md

* removed 'hourstoAdopt'

* removed 'hourstoadopt' from diff

* Restrict JSON schema to not include unspecified properties.

* Fix sample Code.json to match schema.

* update description of laborHours

* add new metadata schema and changelog

* add revised sample_code sons by versions

zachary-kuhn committed:
Revert changes to CSS minification. (#289)

Commit:
Ok new metadataschema (#290)

* started draft schema and sample

* revised version 200

*changed projects to releases
*added field descriptions
*added licenses object
*added permissions object
*added laborHours
*revised measurementType

* removed extra " in schema; formatted diff

* revised exemptionText definition

* updated schema based on github comments

*moved measurementType outside of ‘releases’
*added reusedSourceCode object

* changed reusedSourceCode to relatedCodebase

* added changelog.md

* removed 'hourstoAdopt'

* removed 'hourstoadopt' from diff

* Restrict JSON schema to not include unspecified properties.

* Fix sample Code.json to match schema.

* update description of laborHours

* add new metadata schema and changelog

* add revised sample_code sons by versions

* revisions

*added version
* expanded type for repositoryURL and exemptionText
* removed duplicate changeling

Commit:
Ok new metadataschema (#291)

* started draft schema and sample

* revised version 200

*changed projects to releases
*added field descriptions
*added licenses object
*added permissions object
*added laborHours
*revised measurementType

* removed extra " in schema; formatted diff

* revised exemptionText definition

* updated schema based on github comments

*moved measurementType outside of ‘releases’
*added reusedSourceCode object

* changed reusedSourceCode to relatedCodebase

* added changelog.md

* removed 'hourstoAdopt'

* removed 'hourstoadopt' from diff

* Restrict JSON schema to not include unspecified properties.

* Fix sample Code.json to match schema.

* update description of laborHours

* add new metadata schema and changelog

* add revised sample_code sons by versions

* revisions

*added version
* expanded type for repositoryURL and exemptionText
* removed duplicate changeling

* fixing version descriptions

Commit:
Ok schema200release (#293)

* fixes for version 2.0.0

* changed reusedCodebase->govSoftwareUse

* updated changelog. fixed license reference

* final changes to schema

*rename fields

Commit:
change laborHours type to number from string (#294)

valedaemon committed:
Added search box on home page and search results page.

Commit:
upgrade DOT to green (#297)

Commit:
set additional properties to false for all nested object (#296)

Commit:
adding .codeinventory.yml (#292)

for GSA code inventory process

zachary-kuhn committed:
Create tools for validating and upgrading our code.json schemas. (#298)

zachary-kuhn committed:
Use one source of truth for the schemas.

zachary-kuhn committed:
Merge pull request #299 from presidential-innovation-fellows/feature/updated-search

Feature/updated search

zachary-kuhn committed:
Merge pull request #300 from presidential-innovation-fellows/fix/schema-tools-should-use-published-schema

Use one source of truth for the schemas.

zachary-kuhn committed:
Add the total count and a message if no results are found.

zachary-kuhn committed:
Hide autocomplete on blur, escape, or enter.

zachary-kuhn committed:
We now show agencies in the search results.

zachary-kuhn committed:
Merge pull request #301 from presidential-innovation-fellows/fix/add-no-results-message

Add the total count and a message if no results are found.

zachary-kuhn committed:
Merge pull request #302 from presidential-innovation-fellows/fix/hide-the-autocomplete-more-aggressively

Hide autocomplete on blur, escape, or enter.

zachary-kuhn committed:
Merge pull request #303 from presidential-innovation-fellows/fix/include-agencies-in-autocomplete

We now show agencies in the search results.

zachary-kuhn committed:
Change the blur to behave differently.

zachary-kuhn committed:
Change the blur to behave differently.

Commit:
Merge remote-tracking branch 'origin/fix/autocomplete-blur-does-not-navigate' into fix/autocomplete-blur-does-not-navigate

# Conflicts:
#	src/app/components/repos-search/repos-search.component.ts

zachary-kuhn committed:
Merge pull request #304 from presidential-innovation-fellows/fix/autocomplete-blur-does-not-navigate

Change the blur to behave differently.

Commit:
update reference to 2.0.0 schema (#306)

zachary-kuhn committed:
Add JSDoc to many of the components.

zachary-kuhn committed:
Merge pull request #307 from presidential-innovation-fellows/chore/add-documentation

Add JSDoc to many of the components.

zachary-kuhn committed:
Add typeahead to the header on non-homepage pages.

zachary-kuhn committed:
Load a larger number of repositories and make changes to the site to accomodate this.

zachary-kuhn committed:
Small tweaks to the banner on the homepage.

zachary-kuhn committed:
Include the class name change.

zachary-kuhn committed:
Merge pull request #309 from presidential-innovation-fellows/feature/change-phrasing-on-home-banner

Small tweaks to the banner on the homepage.

zachary-kuhn committed:
Merge pull request #308 from presidential-innovation-fellows/feature/display-all-repos

Feature/display all repos

zachary-kuhn committed:
Sort repositories on the agency in alphabetical order.

zachary-kuhn committed:
Merge pull request #311 from presidential-innovation-fellows/fix/alphabetize-repositories

Sort repositories on the agency in alphabetical order.

zachary-kuhn committed:
Merge pull request #310 from presidential-innovation-fellows/feature/typeahead-search-in-header

Add typeahead to the header on non-homepage pages.

zachary-kuhn committed:
Add the default repositories for ones who do not have code.json files up yet.

zachary-kuhn committed:
Merge pull request #313 from presidential-innovation-fellows/fix/add-default-repositories

Add the default repositories for ones who do not have code.json files…

zachary-kuhn committed:
Add many new UI components to the site, including nav drop down and side nav.

zachary-kuhn committed:
Update wording to be explicit about thousands.

zachary-kuhn committed:
Merge pull request #318 from presidential-innovation-fellows/fix/update-to-thousands

Update wording to be explicit about thousands.

Commit:
Add more indentation to schema bullets
- Added more indentation to the schemas field list in our docs section.

zachary-kuhn committed:
Merge pull request #317 from presidential-innovation-fellows/feature/add-mobile-menu

Feature/add mobile menu

zachary-kuhn committed:
Redesign the home page.

zachary-kuhn committed:
Update the report and fix mobile styling.

zachary-kuhn committed:
Merge pull request #321 from presidential-innovation-fellows/feature/update-report

Feature/update report

zachary-kuhn committed:
Added the Department of Justice to the site.

zachary-kuhn committed:
Update the margin left to center the search box properly.

zachary-kuhn committed:
Update the copy on the site.

zachary-kuhn committed:
Merge pull request #323 from presidential-innovation-fellows/fix/update-copy

Update the copy on the site.

zachary-kuhn committed:
Merge pull request #324 from presidential-innovation-fellows/fix/adjust-centering-on-homepage

Update the margin left to center the search box properly.

zachary-kuhn committed:
Merge pull request #322 from presidential-innovation-fellows/feature/add-DOJ

Added the Department of Justice to the site.

Commit:
Changed usageType to openSource

zachary-kuhn committed:
Update the report for the week of Oct 23.

zachary-kuhn committed:
Merge pull request #338 from GSA/chore/update-report-for-Oct-23

Update the report for the week of Oct 23.

zachary-kuhn committed:
Merge pull request #337 from GSA/feature/redesigned-home

Feature/redesigned home

zachary-kuhn committed:
Update the site with a new set of releases.

zachary-kuhn committed:
Merge pull request #341 from GSA/feature/update-releases

Update the site with a new set of releases.

zachary-kuhn committed:
Various changes to improve initial load time.

zachary-kuhn committed:
Update repos.

zachary-kuhn committed:
Merge pull request #343 from GSA/feature/update-repos

Update repos.

Commit:
Change CodeClimate test reporter.
- Removed old CodeClimate test reporter and added code to use their new version.

Commit:
Silly change to test CircleCI webhook

zachary-kuhn committed:
Fix the test suite.

zachary-kuhn committed:
Update styling of the search results page.

zachary-kuhn committed:
Update to the most recent release updates.

zachary-kuhn committed:
Merge pull request #349 from GSA/chore/update-releases

Update to the most recent release updates.

Commit:
Add robots.txt webpack plugin

zachary-kuhn committed:
Remove the next three years because time has passed.

Commit:
Add robots.txt webpack plugin config
- Added robotstxt-webpack-plugin config
- Added robotstxt-webpack-plugin to require statements
- Added robotstxt-webpack-plugin to webpack plugins config
- Removed github deploy code.
- Cleaned up process flag validations.

Commit:
Remove config/webpack.github-deploy.js
- We should not be building for github pages. GH pages are not authorized for use.

Commit:
Uncomment deployment code.
- Left some important code commented. It was caught during review!

zachary-kuhn committed:
Scroll down based on the offset of the header height.

zachary-kuhn committed:
Add a comment about what the function does.

zachary-kuhn committed:
Merge pull request #352 from GSA/fix/mobile-not-scrolling-down-to-header

Scroll down based on the offset of the header height.

zachary-kuhn committed:
Merge pull request #350 from GSA/feature/copy-updates

Remove the next three years because time has passed.

zachary-kuhn committed:
Merge pull request #346 from GSA/fix/fix-test-suite

Fix the test suite.

BalajiJBcs committed:
Metadata update

Metadata update, added Site title and site name under the standard of the open graph protocol.

RicardoAReyes committed:
Schema 2.0.0 Documentation update to improve language and add exemptByPolicyDate exception option.

zachary-kuhn committed:
Update the Circle CI config to use the new CodeClimate process.

zachary-kuhn committed:
Merge pull request #355 from GSA/fix/fix-circle-ci-config

Update the Circle CI config to use the new CodeClimate process.

zachary-kuhn committed:
Merge branch 'master' into patch-1

zachary-kuhn committed:
Merge branch 'master' into feature/schema-doc-update

zachary-kuhn committed:
Update releases for 12/4/2017.

zachary-kuhn committed:
Show all releases.

zachary-kuhn committed:
Merge pull request #354 from GSA/feature/schema-doc-update

Schema 2.0.0 Documentation update to improve language and add exemptB…

zachary-kuhn committed:
Merge pull request #356 from GSA/chore/update-releases

Chore/update releases

zachary-kuhn committed:
Revert to stable dependencies.

zachary-kuhn committed:
Merge pull request #357 from GSA/fix/new-dependency-breaks-app

Revert to stable dependencies.

zachary-kuhn committed:
Merge pull request #353 from BalajiJBcs/patch-1

Metadata update

zachary-kuhn committed:
Update documentation to match validation.

BalajiJBcs committed:
Added Twitter cards

When users share URL on twitter it will display as Twitter cards with URL, images, description, and title than just URL.

zachary-kuhn committed:
Merge pull request #358 from GSA/fix/update-documentation-for-related-code

Update documentation to match validation.

BalajiJBcs committed:
Merge branch 'master' into patch-1

RicardoAReyes committed:
Google Analytics account reset for existing property id

RicardoAReyes committed:
Merge branch 'master' into feature/google-analytics-account-reset

zachary-kuhn committed:
Merge pull request #361 from GSA/feature/google-analytics-account-reset

Google Analytics account reset for existing property id

RicardoAReyes committed:
moving ga reset file to meta directory

BalajiJBcs committed:
Merge branch 'master' into patch-1

zachary-kuhn committed:
Merge branch 'master' into feature/ga-account-reset-meta

zachary-kuhn committed:
Merge pull request #364 from GSA/feature/ga-account-reset-meta

moving ga reset file to meta directory

zachary-kuhn committed:
Merge pull request #363 from GSA/fix/improve-load-times

Fix/improve load times

zachary-kuhn committed:
Show the agency properly on the search results page.

zachary-kuhn committed:
Merge pull request #365 from GSA/fix/agencies-throw-error-on-search

Show the agency properly on the search results page.

jcastle-zz committed:
Roadmap (#367)

* created roadmap branch for dev of Roadmap for Code.gov

* updates to roadmap page, need another look and some additional content

zachary-kuhn committed:
Add our icons folder.

zachary-kuhn committed:
Merge pull request #371 from GSA/fix/icons-were-ignored

Add our icons folder.

zachary-kuhn committed:
Add additional checks to make sure no errors are thrown. (#372)

BalajiJBcs committed:
Merge branch 'master' into patch-1

RicardoAReyes committed:
Latest releases update, Dec 15. 2017

RicardoAReyes committed:
Merge pull request #374 from GSA/feature/update-repos

Latest releases update, Dec 15. 2017

zachary-kuhn committed:
Update production deploy branch.

zachary-kuhn committed:
Merge branch 'master' into patch-1

zachary-kuhn committed:
Merge pull request #359 from BalajiJBcs/patch-1

Added Twitter cards

zachary-kuhn committed:
Change in Lodash imports introduced a bug in the upgrader.

zachary-kuhn committed:
Update the schemas used on the web to allow exemptByPolicyDate.

zachary-kuhn committed:
Merge branch 'master' into fix/change-federalist-pages-to-federalist-prod

zachary-kuhn committed:
Merge pull request #375 from GSA/fix/change-federalist-pages-to-federalist-prod

Update production deploy branch.

zachary-kuhn committed:
Merge branch 'master' into fix/improper-lodash-import-causing-error

zachary-kuhn committed:
Merge branch 'master' into fix/allow-exempt-by-policy-date

zachary-kuhn committed:
Merge pull request #377 from GSA/fix/improper-lodash-import-causing-error

Change in Lodash imports introduced a bug in the upgrader.

zachary-kuhn committed:
Merge branch 'master' into fix/allow-exempt-by-policy-date

zachary-kuhn committed:
Merge pull request #378 from GSA/fix/allow-exempt-by-policy-date

Update the schemas used on the web to allow exemptByPolicyDate.

jcastle-zz committed:
Roadmap (#379)

Updates to roadmap - fixed table widths, added banner, and h1 content and styles

Commit:
Remove .git from https regular expression.
- https URL to repos do not have the .git suffix. This made the matching of the remote repository not work.

Commit:
Remove schema date format.
- Our schema updater sets empty dates to empty strings. This will make code.json fail when passed through the validator with unchanged date values. The short term solution is to remove the format check in the schema so that the validator accepts empty strings.

Commit:
Update releases with harvester validation changes

Commit:
Change clean script
-- npm cache clean is depricated verify should be used.

Commit:
Change member ordering rule.
- the member ordering rule was incorrectly set. The correct configuration was added.

Commit:
Remove --type-check option from npm script.
= The --type-check option was depricated from tslint. The --project options is the only one needed.

Commit:
Fix linter errors
- Fixed multiple linter errors

Commit:
Package.json cleanup
- Changed the repository URL to the current one.
- Changed the bugs URL to the current one.
- Changed the author to Code.gov team.
- Added project contributors.
- Fixed spacing issues.

Commit:
Refactor circleci.yml
- Updated circleci config to version 2
- Added workflows
- Added test, integration and deploy jobs
- updated the docker images to 6.12.2
- Added new jobs to workflow.

Commit:
Add deploy to prod and rename staging deploy.
- Renamed deploy job to deploy-staging
- Added deploy-prod job
- Added deploy-prod to workflow. This will only execute with tags and ignores all branches.
- Changed the docker image used for deploy-staging.

Commit:
Fix contributor url typo

Commit:
Add deployment hold job.
- Added deploy-hold job. Main reason is that we are not 100% ready for continuos deployment. There are some branches that are being used in staging for testing purposes. This should change shortly.
- Added deploy-hold requirement to deploy-staging and deploy-production jobs.

Commit:
Trying a fix to configuration

Commit:
Fix yml structure bug

Commit:
Add filters to hold job in workflow

Commit:
Reverting WORKING_DIRECTORY change.
- The environment variable is not playing nice.

Commit:
Removing the remaining WORKING_DIRECTORY environment variables.

Commit:
Updated http-server dependency
- One of http-server's dependencies (ecstatic) was flagged as having a major vulnerability. Once http-server was updated ecstatic was also updated out of the version with the vulnerability.

Commit:
Updated dependencies.

BalajiJBcs committed:
Merge pull request #1 from GSA/master

Pull updates from GSA

Commit:
Image optimization

The 4 image is compressed without losing any quality of image which reduced 1MB of file size which saves 1MB data and faster loading of site.

Commit:
Image optimization

The 4 image is compressed without losing any quality of image which reduced 1MB of file size which saves 1MB data and faster loading of site.

BalajiJBcs committed:
Security Updates

@Zachary-Kuhn , I'm sorry to bring this to your concern. I see the following file is not supposed to be here. 
I don't know whether it as some use in Code.gov. But as far my knowledge it is a small human error where you saved your personnel file which everyone does. So I removed the file. If I understand wrongly and removed that pdf. Pardon me for it Mr.Zachry-Kuhn.

BalajiJBcs committed:
Merge branch 'master' into master

BalajiJBcs committed:
Merge branch 'master' into master

froi commented:
Hi @BalajiJBcs 

Thank you so much for your continued help with this project. You are correct that this file is not part of the project. I went ahead and removed the file from our git history. I'll circle back once I triage this. 

Thanks again!

BalajiJBcs committed:
Deletion of Some file - Security

I exactly don't know, why the file is here. But I feel suspicious that this file contains some login credentials including the password. But I am not quite sure I'm right or wrong. Please remove this file as a security update and also change the password for the account. @RicardoAReyes if I'm made the wrong alert pardon me.

BalajiJBcs committed:
Merge branch 'master' into master

BalajiJBcs committed:
Merge branch 'master' into master

BalajiJBcs committed:
Merge branch 'master' into master

BalajiJBcs committed:
Merge branch 'master' into master

saracope commented:
Closing out as this repo is being archived. Please check out the new repo at [code-gov-front-end](https://github.com/GSA/code-gov-front-end).
